### PR TITLE
test(extension): hd wallet first 3 tests

### PR DESCRIPTION
**Pull request by oldGreg5**
# Checklist

- [ ] JIRA [LW-7415](https://input-output.atlassian.net/browse/LW-7415)
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

First 3 HD wallet tests. Checking discovery of multiple used addresses and correctness of transaction history when different wallet addresses were used.

[LW-7543]: https://input-output.atlassian.net/browse/LW-7543?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[LW-7415]: https://input-output.atlassian.net/browse/LW-7415?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/1476/5859211292/index.html) for [ff353bff](https://github.com/input-output-hk/lace/pull/386/commits/ff353bffe40b19f8a8da5c65af7535ccc7c51d7d)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 33     | 2      | 0       | 0     | 35    | ❌     |

### CODE REVIEW

**Review by ljagiela — APPROVED** [review]

File: packages/e2e-tests/src/steps/nftsExtendedSteps.ts [hunk 1]
```diff
@@ -30,6 +30,10 @@ Given(/^I'm sending an NFT with name: "([^"]*)"$/, async (nftName: string) => {
   await nftsPageObject.progressWithSendUntilPasswordPage(nftName);
 });
 
+Given(/^I'm sending an NFT with name: "([^"]*)" with HD wallet$/, async (nftName: string) => {
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/steps/nftsExtendedSteps.ts:33]: Let's use article `the` here

File: packages/e2e-tests/src/steps/nftsCommonSteps.ts [hunk 2]
```diff
@@ -85,6 +85,19 @@ Given(
   }
 );
 
+Given(
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/steps/nftsCommonSteps.ts:88]: It seems that we have 2 almost identical steps - this one and the one starting in L75.
Please consider merging them together.
I'd also suggest a little renaming:
`/^I use a (wallet_type_1|wallet_type_2) wallet with "([^"]*)" NFT in (popup|extended) mode$/`
Don't forget to adjust the wallet type names and parameter names and order in step definition.

File: packages/e2e-tests/src/features/HdWalletExtended.feature [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  Background:
+    Given Lace is ready for test
+
+  @LW-7550
+  Scenario: Tokens - HD wallet assets discovered in Lace
+    Then I see total wallet balance in ADA is "10019.03"
+    And I see tMin token with the ADA balance of "21"
+    When I navigate to NFTs extended page
+    And I see NFT with name: "DEV 3432" on the NFTs page
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/features/HdWalletExtended.feature:12]: Please replace `And` with `Then`

File: packages/e2e-tests/src/features/HdWalletExtended.feature [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    And I see NFT with name: "$rinodino" on the NFTs page
+
+  @LW-7552
+  Scenario Outline: Transactions - HD wallet transactions displayed correctly
+    When I navigate to Transactions extended page
+    And I can see transaction <txNumber> has type "<txType>" and value "<txValue>"
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/features/HdWalletExtended.feature:18]: As above

File: packages/e2e-tests/src/assert/tokensPageAssert.ts [hunk 5]
```diff
@@ -230,6 +230,22 @@ class TokensPageAssert {
     }
   }
 
+  async expectedAdaBalance(balanceInAda: number) {
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/assert/tokensPageAssert.ts:233]: Let's keep previous naming convention where all assertion names start with `assert...`

File: packages/e2e-tests/src/assert/tokensPageAssert.ts [hunk 6]
```diff
@@ -230,6 +230,22 @@ class TokensPageAssert {
...
+    await this.expectedTokenBalance(0, balanceInAda);
+  }
+
+  async expectedTMinBalance(balance: number) {
+    const TMinIndex = await TokensPage.getTokenRowIndex('tMIN');
+    await this.expectedTokenBalance(TMinIndex, balance);
+  }
+
+  async expectedTokenBalance(tokenIndex: number, tokenBalance: number) {
+    const balance: number = await TokensPage.tokenBalance(tokenIndex)
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/assert/tokensPageAssert.ts:243]: I'd suggest to not chain promises with `then(...)` and use 
```
const balance = Number((await TokensPage.tokenBalance(tokenIndex).getText()).replace(',', ''));
```
instead

**Review by wklos-iohk — CHANGES_REQUESTED** [review]

**Review by wklos-iohk — APPROVED** [review]

**Pull request merged**